src/client.ts

```diff
@@ -731,6 +735,17 @@ export class StreamChat<
     } = {},
   ): Promise<T> => {
     await this.tokenManager.tokenReady();
+
+    const resetTime = this.rateLimitResetTime[url]
```

Comment:
how is old reset time cleaned ?

Reply:
Line 748

Reply:
throw prevents it

Reply:
Yeah I mean ... we need it in case of error. If user calls the api again, it should again throw error. We should only cleanup if resetTime has expired